### PR TITLE
Add list of reserved variable names

### PR DESCRIPTION
**Pull request by Odonno**
I have included the list of reserved variable names as this is a common question.

1. I have added this list to the `parameters` page. It may not be the best place for it. What do you think?
2. I have only listed the variables I know. It may not be exhaustive. Are there any other variable names that I should know?

### CODE REVIEW

File: app/templates/docs/surrealql/parameters.hbs [hunk 1]
```diff
@@ -31,6 +31,55 @@
...
+					<td>
+						<code>$token</code>
+					</td>
+					<td>Used for authentication</td>
+				</tr>
+				<tr>
+					<td>
+						<code>$session</code>
+					</td>
+					<td>Used for authentication</td>
```
Comment (kearfy) [inline app/templates/docs/surrealql/parameters.hbs:56]:
```suggestion
					<td>Represents values from the <Link @link="docs.surrealql.functions.session">session</Link> functions as an object</td>
```

File: app/templates/docs/surrealql/parameters.hbs [hunk 2]
```diff
@@ -31,6 +31,55 @@
...
+					<th w-20>Name</th>
+					<th w-60>Description</th>
+				</tr>
+			</thead>
+			<tbody>
+				<tr>
+					<td>
+						<code>$token</code>
+					</td>
+					<td>Used for authentication</td>
```
Comment (kearfy) [inline app/templates/docs/surrealql/parameters.hbs:50]:
```suggestion
					<td>Represents values held inside the JWT token used for the current session</td>
```

File: app/templates/docs/surrealql/parameters.hbs [hunk 3]
```diff
@@ -31,6 +31,55 @@
...
+						<code>$before</code>
+					</td>
+					<td>Used on table/event definitions</td>
+				</tr>
+				<tr>
+					<td>
+						<code>$after</code>
+					</td>
+					<td>Used on table/event definitions</td>
+				</tr>
```
Comment (kearfy) [inline app/templates/docs/surrealql/parameters.hbs:75]:
```suggestion
				<tr>
					<td>
						<code>$before</code>
					</td>
					<td>Represents the value before a mutation on a field</td>
				</tr>
				<tr>
					<td>
						<code>$after</code>
					</td>
					<td>Represents the value after a mutation on a field</td>
				</tr>
				<tr>
					<td>
						<code>$value</code>
					</td>
					<td>Represents the value after a mutation on a field (identical to <code>$value</code></td>
				</tr>
```

File: app/templates/docs/surrealql/parameters.hbs [hunk 4]
```diff
@@ -31,6 +31,55 @@
...
+			</tbody>
```
Comment (kearfy) [inline app/templates/docs/surrealql/parameters.hbs:76]:
```suggestion
				<tr>
					<td>
						<code>$parent</code>
					</td>
					<td>Represents the parent record in a subquery</td>
				</tr>
			</tbody>
```

File: app/templates/docs/surrealql/parameters.hbs [hunk 5]
```diff
@@ -31,6 +31,55 @@
...
+	<p>SurrealDB uses predefined variables. For that purpose, you can use those variables inside your query but you cannot declare new parameters using one of the following name:</p>
+	<Layout::Table filled>
+		<table>
+			<thead>
+				<tr>
+					<th w-20>Name</th>
+					<th w-60>Description</th>
+				</tr>
+			</thead>
+			<tbody>
```
Comment (kearfy) [inline app/templates/docs/surrealql/parameters.hbs:45]:
```suggestion
			<tbody>
				<tr>
					<td>
						<code>$auth</code>
					</td>
					<td>Represents the currently authenticated scope user</td>
				</tr>
```

**Review by kearfy — CHANGES_REQUESTED** [review]

File: app/templates/docs/surrealql/parameters.hbs (hunk 3 above)
Comment (Odonno) [reply app/templates/docs/surrealql/parameters.hbs]: `$value` identical to `$value`? Are you sure? :)

So I assume you meant `$after`?

Comment (kearfy) [reply app/templates/docs/surrealql/parameters.hbs]: Yes! My bad 🤪

Comment (kearfy) [reply app/templates/docs/surrealql/parameters.hbs]: I will make some new suggestions, one second

**Comment by Odonno** [comment]
I was thinking about developer usage. I suppose `$before`, `$after` and `$value` are only used for `DEFINE` statement. I don't know if it throws an error on another statements like `SELECT`. 

If it does not throw error, I was thinking about adding a `context` column to know where we cannot use those variables.

**Comment by Odonno** [comment]
Also add type for each variable.

File: app/templates/docs/surrealql/parameters.hbs [hunk 6]
```diff
@@ -31,6 +31,73 @@
...
+						<code>$after</code>
+					</td>
+					<td>Represents the value after a mutation on a field</td>
+				</tr>
+				<tr>
+					<td>
+						<code>$value</code>
+					</td>
+					<td>Represents the value after a mutation on a field (identical to <code>$value</code>)</td>
+				</tr>
```
Comment (kearfy) [inline app/templates/docs/surrealql/parameters.hbs:87]:
```suggestion
				<tr>
					<td>
						<code>$before</code>
					</td>
					<td>Represents the value before a mutation on a field</td>
				</tr>
				<tr>
					<td>
						<code>$after</code>
					</td>
					<td>Represents the value after a mutation on a field</td>
				</tr>
				<tr>
					<td>
						<code>$value</code>
					</td>
					<td>Represents the value after a mutation on a field (identical to <code>$after</code> in the case of an event)</td>
				</tr>
				<tr>
					<td>
						<code>$input</code>
					</td>
					<td>Represents the initially inputted value in a field definition, as the <code>value</code> clause could have modified the <code>$value</code> variable</td>
				</tr>				
```

**Review by kearfy — APPROVED** [review]

**Pull request merged**